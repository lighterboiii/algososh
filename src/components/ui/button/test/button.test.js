import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from '../button';

describe('Тест компонента Button:', () => {
  it('Компонент Button с текстом отрисован корректно', () => {
    const button = renderer.create(<Button text='Новый массив' />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it('Компонент Button без текста отрисован корректно', () => {
    const button = renderer.create(<Button text='' />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it('Компонент Button в заблокированном состоянии отрисован корректно', () => {
    const button = renderer.create(<Button disabled />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it('Компонент Button в состоянии загрузки отрисован корректно', () => {
    const button = renderer.create(<Button isLoader />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it('Корректно выполняется коллбек при клике на компонент Button', () => {
    window.alert = jest.fn();
    render(<Button text='Коллбек' onClick={() => alert('Протестировано')} />);
    const button = screen.getByText('Коллбек');
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith('Протестировано');
  })
});
