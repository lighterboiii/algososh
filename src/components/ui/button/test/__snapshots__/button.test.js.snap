// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента Button: Компонент Button без текста отрисован корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`Тест компонента Button: Компонент Button в заблокированном состоянии отрисован корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест компонента Button: Компонент Button в состоянии загрузки отрисован корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тест компонента Button: Компонент Button с текстом отрисован корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Новый массив
  </p>
</button>
`;
